Data/HashMap/Base.hs

```diff
@@ -252,7 +266,9 @@ instance (Eq k, Eq v) => Eq (HashMap k v) where
 
 equal :: (k -> k' -> Bool) -> (v -> v' -> Bool)
       -> HashMap k v -> HashMap k' v' -> Bool
-equal eqk eqv t1 t2 = go (toList' t1 []) (toList' t2 [])
+equal eqk eqv (HashMap s1 t1) (HashMap s2 t2)
+    | s1 == s2  = go (toList' t1 []) (toList' t2 [])
+    | otherwise = False
```

Comment:
You can write this as `s1 == s2 && go ...`.

Reply:
Nice find, thanks.